Issue 508 - Update Location Services

heyjamesknight opened the pull request:
- Added LocationClient
    - Hides old Android/new Play Services implementations behind a shared Interface.
    - Moved old Location code into AndroidLocationClient.
    - Created new GoogleLocationClient which uses Play Service's FusedLocationProvider to provide Location updates.
- Updated Location Activities to use the new LocationClient
    - GeoPointActivity, GeoPointMapActivity, GeoPointOsmMapActivity, GeoShapeGoogleMapActivity, GeoTraceGoogleMapActivity, GeoTraceOsmMapActivity were updated.
    - GeoShapeOsmActivity was left using the legacy code, as the new LocationClient does not provide a way to determine if specific LocationProviders are available.
- Added Unit Tests for LocationClient implementations.
- Added Robolectric tests for Activities.

shobhitagarwal1612 committed:
Refactors code

shobhitagarwal1612 committed:
Updates emptyView on list update in FormDownloadList

shobhitagarwal1612 committed:
Sets custom toolbar in FormEntryActivity

shobhitagarwal1612 committed:
Adds toolbar to preferences

shobhitagarwal1612 committed:
Removes unused imports

shobhitagarwal1612 committed:
Adds toolbar to GoogleDriveActivity

shobhitagarwal1612 committed:
Fixes TestFails

shobhitagarwal1612 committed:
Replaces with supportMethods

shobhitagarwal1612 committed:
Modifies text label and buttons style in main_menu

shobhitagarwal1612 committed:
Makes default text color black

shobhitagarwal1612 committed:
textAll caps in buttons

shobhitagarwal1612 committed:
Adds toobar to FormHierarchyActivity

shobhitagarwal1612 committed:
Adds navigation icon in toolbar

shobhitagarwal1612 committed:
Adds accentColor

shobhitagarwal1612 committed:
Adds accentColor

shobhitagarwal1612 committed:
Removes redundant theme-v11

shobhitagarwal1612 committed:
Adds a separate theme for v21 and above devices

shobhitagarwal1612 committed:
Adds support design dependency in build.gradle

Adds custom shadow resources

shobhitagarwal1612 committed:
Adds shadow to main_menu

shobhitagarwal1612 committed:
Adds shadow to chooser_list_layout

shobhitagarwal1612 committed:
Adds shadow to instance_uploader_list

shobhitagarwal1612 committed:
Adds shadow to drive_layout

shobhitagarwal1612 committed:
Adds shadow to remote_file_manage_list

shobhitagarwal1612 committed:
Adds shadow to file_manager_layout

shobhitagarwal1612 committed:
Adds shadow to preferences_layout

shobhitagarwal1612 committed:
Adds shadow to heirarchy_layout

shobhitagarwal1612 committed:
Adds shadow to form_entry

shobhitagarwal1612 committed:
Fixes bugs

shobhitagarwal1612 committed:
Reverts back changes to SplashScreenActivity

shobhitagarwal1612 committed:
Fixes bugs

shobhitagarwal1612 committed:
Load form controller before destroying task (#1013)

grzesiek2010 committed:
Fix crash when accessing jump menu (#1015)

shobhitagarwal1612 committed:
Fixes navigation icon in Delete Saved Form

shobhitagarwal1612 committed:
Fixes crash while device rotation

shobhitagarwal1612 committed:
Bugs fixed

shobhitagarwal1612 committed:
Removes unused imports

shobhitagarwal1612 committed:
Configure Server Settings by scanning 2D barcode #781 (#843)

* Adds a fragment in AdminPreferences to display QRCode for dummy text

* Adds preferences data to the QRCode

* Adds button to scan the QR code

* Applies imported settings to shared preferences

* Adds username and password to the QRCode

* Adds a fallback in case the maximum character limit is reached for encoding
Saves preferences for protocol

* Compresses jsonObject before encoding into QRCode

* Integrates embedded scanner

* Adds share action to options menu

* Adds button to select image from SD Card

* Saves bitmap to cache for sharing instead of sd card

* Refactors error handling for generating code
Updates qr code after importing settings

* Adds a utils class to import settings from JSON object

* Adds GenerateQRCode async task and interface

* Fixes fragment overlay bug while rotating device

* Returns to landing screen on successful import

* Creates QRCode using all modified settings

* Imports settings from QRCode

* Sets default value for unmodified settings on import

* Adds dialog to choose admin and server password

* Adds autosend wifi and network keys to the general keys

yanokwa committed:
Typo fix for invalid barcode (#1024)

yanokwa committed:
Move building of QR code image closer to where it's used (#1025)

shobhitagarwal1612 committed:
Merge branch 'master' into issue-493

# Conflicts:
#	collect_app/build.gradle
#	collect_app/src/main/AndroidManifest.xml
#	collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesActivity.java

yanokwa committed:
Add required File import (#1026)

grzesiek2010 committed:
Fix NullPointerException in FormEntryActivity #1014 (#1018)

shobhitagarwal1612 committed:
Displays empty message in case no item found in FormList/InstanceList/ViewSent

shobhitagarwal1612 committed:
Merge branch 'master' into issue-493

MukundAnanthu committed:
Space out action results - #730 (#1019)

shobhitagarwal1612 committed:
Displays options menu only if the device has hardOptionsKey

shobhitagarwal1612 committed:
Adds menu

shobhitagarwal1612 committed:
Fixes admin password not setting by QR code scan (#1027)

shobhitagarwal1612 committed:
Updates test cases

shobhitagarwal1612 committed:
Adds checkstyle MemberName (#989)

shobhitagarwal1612 committed:
Merge branch 'master' into issue-493

# Conflicts:
#	collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java
#	collect_app/src/main/java/org/odk/collect/android/activities/BearingActivity.java
#	collect_app/src/main/java/org/odk/collect/android/activities/FormChooserList.java
#	collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadList.java
#	collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java
#	collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java
#	collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java
#	collect_app/src/main/java/org/odk/collect/android/activities/GoogleSheetsUploaderActivity.java
#	collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java
#	collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderActivity.java

shobhitagarwal1612 committed:
Fixes checkstyle issues
Removes blank file GoogleSheetsTask

MukundAnanthu committed:
Only parse response message if OpenRosaResponse tag exists (#1031)

grzesiek2010 committed:
Replaces Existing Data on Google Spreadsheet (#1029)

grzesiek2010 committed:
Field-list appears when deleting repeat with no field-list (#1034)

shobhitagarwal1612 committed:
Crashes in FormEntryActivity fixed

shobhitagarwal1612 committed:
Adds sort and filter menu to ApplistActivity

shobhitagarwal1612 committed:
Reduces the gap between title and navigation icon in toolbar

shobhitagarwal1612 committed:
Removes blank file (#1037)

grzesiek2010 committed:
Send button should be disabled when all form are unselected (#1046)

MukundAnanthu committed:
Use embedded zxing lib in BarcodeWidget. Fixes #1021 (#1035)

* Use embedded zxing lib in BarcodeWidget. Fixes #1021

* Removed unused imports to conform to checkstyle tests. Fixes #1021

* Added code for handling the result from Zxing barcode scanning lib

* Ensure refreshCurrentView() is called before returning from barcode scan result processing.

* Removed unused string with label barcode_scanner_error from string.xml

lognaturel committed:
Remove redundant Hindi translation. (#1049)

yanokwa committed:
Add support for snapshots of JavaRosa (#1050)

shobhitagarwal1612 committed:
Refactor shared preferences (#1036)

* Refactors shared preferences

* Removes unnecessary loop

* Copy keySets so they can be modified.

* Add missing general preferences to QR code.

grzesiek2010 committed:
Add selfie mode (#1016)

grzesiek2010 committed:
Removed unnecessary brackets (#1055)

shobhitagarwal1612 committed:
Update locale on importing language from QR code (#1058)

MukundAnanthu committed:
Fixes incorrect time conversion from minutes to seconds. Fixes #1053. (#1054)

lognaturel committed:
Don't require camera features. (#1060)

shobhitagarwal1612 committed:
Groups user access control in admin settings

shobhitagarwal1612 committed:
Adds summary and string resources

shobhitagarwal1612 committed:
Groups general settings and adds icons

shobhitagarwal1612 committed:
Refactors code
Removes redundant code

shobhitagarwal1612 committed:
Updates code for hiding user settings items

grzesiek2010 committed:
SelectOne and SelectMulti widgets don't work well with compact-x appearance (#1062)

yanokwa committed:
JavaRosa 2.2.0 (#1068)

lognaturel committed:
Update translations from Transifex. (#1069)

lognaturel committed:
Fix Italian translation apostrophes. (#1070)

shobhitagarwal1612 committed:
Merge branch 'master' into settings-redesign

grzesiek2010 committed:
NonEmptyAtclauseDescription (#1083)

grzesiek2010 committed:
Place cursor at the end of text in EditText (#1081)

grzesiek2010 committed:
Add ApplySharedPref lint (#1079)

grzesiek2010 committed:
Remove daylight savings gap skip (#1077)

grzesiek2010 committed:
Fall back to open field when external app missing (#1076)

alxndrsn committed:
Fix indentation of strings.xml (#1073)

alxndrsn committed:
Remove unused code (#1071)

shobhitagarwal1612 committed:
Merges autosend wifi and cellular to a list preference

MukundAnanthu committed:
Fix typos and localize variables in geo activities (#1056)

grzesiek2010 committed:
ODK crashes after using geotrace osm with an automatic mode (#1091)

meletis committed:
Use tiles table when loading offline map tiles. (#1087)

grzesiek2010 committed:
Added message for empty view sent form list (#1088)

shobhitagarwal1612 committed:
Adds toolbar to open source licenses screen

shobhitagarwal1612 committed:
Improves visibility of disabled button in audio and video widget

shobhitagarwal1612 committed:
Fixes icons in GeoPointMapActivity

shobhitagarwal1612 committed:
Merge branch 'master' into issue-493

# Conflicts:
#	collect_app/src/main/java/org/odk/collect/android/activities/GeoPointMapActivity.java

shobhitagarwal1612 committed:
Updates support library version

shobhitagarwal1612 committed:
Improved grouping in user interface

shobhitagarwal1612 committed:
Adds user and device identity category in general settings

shobhitagarwal1612 committed:
Renames preference titles

shobhitagarwal1612 committed:
Removes secondary summary from user access settings in admin settings

grzesiek2010 committed:
Exception when uploading instances (#1047)

#975 suggests more invasive replacements. `discardEntityBytes` is not well-understood and probably isn't necessary.

thedavidmccann committed:
Escape html characters from markdown before conversion (#1044)

shobhitagarwal1612 committed:
Modifies the behaviour of platform preference

shobhitagarwal1612 committed:
Removes unused resources

shobhitagarwal1612 committed:
FIxes show/hide server preference from admin settings

shobhitagarwal1612 committed:
Deletes unused code

shobhitagarwal1612 committed:
Adds license

grzesiek2010 committed:
Fixed DateWidget

grzesiek2010 committed:
Fixed TimeWidget

grzesiek2010 committed:
Fixed bug in DateTimeWidget

shobhitagarwal1612 committed:
Removes unused imports

shobhitagarwal1612 committed:
Refactor code

shobhitagarwal1612 committed:
Fix crash in form download list

shobhitagarwal1612 committed:
Adds fonts

shobhitagarwal1612 committed:
Fix lint issues

shobhitagarwal1612 committed:
Adds new icons
Refactors DisabledPreferencesRemover class

shobhitagarwal1612 committed:
Renames the drawables more accurately

shobhitagarwal1612 committed:
Refactors code

shobhitagarwal1612 committed:
Merge branch 'master' into settings-redesign

# Conflicts:
#	collect_app/src/main/res/values/strings.xml

shobhitagarwal1612 committed:
Renames string use_device_language to use_phone_language

yanokwa committed:
Link to forum rather than mailing lists (#1102)

lognaturel committed:
Merge pull request #912 from shobhitagarwal1612/issue-493

Migrating to Android Support Library

shobhitagarwal1612 committed:
Merge branch 'master' into settings-redesign

# Conflicts:
#	collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java
#	collect_app/src/main/java/org/odk/collect/android/preferences/AboutPreferencesActivity.java
#	collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java
#	collect_app/src/main/java/org/odk/collect/android/preferences/GooglePreferencesActivity.java
#	collect_app/src/main/java/org/odk/collect/android/preferences/OtherPreferencesActivity.java
#	collect_app/src/main/java/org/odk/collect/android/preferences/PreferencesActivity.java

shobhitagarwal1612 committed:
Adds check for disabling preferenceGroups having defined child preferences

shobhitagarwal1612 committed:
Adds toolbar to ServerSettings

shobhitagarwal1612 committed:
Adds toolbar for nested preference screens

grzesiek2010 committed:
Localize response when default Aggregate response is given (#1095)

alxndrsn committed:
Close stream safely in WebUtils.discardEntityBytes() (#1103)

Issue: #975

shobhitagarwal1612 committed:
Adds backward compatibility for autosend setting

meletis committed:
Don't place a marker if location hasn't been received yet (#1112)

shobhitagarwal1612 committed:
Fixes layout of show_qrcode_fragment

shobhitagarwal1612 committed:
Adds backward compatibility for importing settings when app starts

shobhitagarwal1612 committed:
Fixes font size of preferences

shobhitagarwal1612 committed:
modfies textSize form 17sp to 18sp in settings

shobhitagarwal1612 committed:
Removes manually updating the emptyView (#1119)

yanokwa committed:
Add forum to about screen, improve website link copy (#1118)

shobhitagarwal1612 committed:
Renames methods

shobhitagarwal1612 committed:
Fixes migration of autosend preferenes from settings file

shobhitagarwal1612 committed:
Adds autosend preference migrator

shobhitagarwal1612 committed:
Fixes disabled preferences remover

grzesiek2010 committed:
Removed unused resources (#1082)

shobhitagarwal1612 committed:
Merge branch 'master' into settings-redesign

# Conflicts:
#	collect_app/src/main/res/values-de/strings.xml
#	collect_app/src/main/res/values-es/strings.xml
#	collect_app/src/main/res/values-fr/strings.xml
#	collect_app/src/main/res/values-it/strings.xml
#	collect_app/src/main/res/values-ja/strings.xml
#	collect_app/src/main/res/values-km/strings.xml
#	collect_app/src/main/res/values-sv-rSE/strings.xml
#	collect_app/src/main/res/values-th-rTH/strings.xml
#	collect_app/src/main/res/values-uk/strings.xml
#	collect_app/src/main/res/values/strings.xml

shobhitagarwal1612 committed:
Removes unused resources

grzesiek2010 committed:
Database upgrade

grzesiek2010 committed:
Read auto_delete and auto_send from form

grzesiek2010 committed:
Handle autoDelete

grzesiek2010 committed:
Handle autoSubmit

lognaturel committed:
Merge pull request #1101 from shobhitagarwal1612/settings-redesign

Settings redesign

grzesiek2010 committed:
RangeWidget integer no appearance (#1108)

grzesiek2010 committed:
Use floating action button in drawing activities (#961)

grzesiek2010 committed:
Fix items disappearing from View Sent Form (#1126)

grzesiek2010 committed:
Fix delete after send (#1130)

grzesiek2010 committed:
Fixed blocked swiping (#1141)

yanokwa committed:
Add tested devices to README (#1143)

grzesiek2010 committed:
Added Samsung Galaxy S4 (#1146)

grzesiek2010 committed:
Use username/password from externally loaded settings (#1140)

shobhitagarwal1612 committed:
Adds device Xolo Q700s plus to README (#1147)

lognaturel committed:
Use JavaRosa snapshot 2.3.0. (#1148)

jovaee committed:
Added the loading of settings from a JSON file. (#1125)

grzesiek2010 committed:
Fixed bug in InstanceUploaderTask

grzesiek2010 committed:
Fixed link to Transifex (#1149)

shobhitagarwal1612 committed:
Refactors migration for auto send (#1139)

grzesiek2010 committed:
GeoTrace buttons are too large on Android v5.0 and v6.0 (#1155)

grzesiek2010 committed:
Use full landscape screen when gathering signature (#1152)

cdwuestner committed:
Rename language option from "phone language" to "device language" (#1159)

shobhitagarwal1612 committed:
Fixes bug : 'Go Up' option looks like an active button (#1157)

shobhitagarwal1612 committed:
Removes error level logs for normal conditions (#1164)

yanokwa committed:
Trim about screen summaries (#1162)

shobhitagarwal1612 committed:
Improves label, list and listmulti widgets (#1166)

* Improves label, list and list multi widgets

* label color --> black
font size --> questionFont size

* Adds XML for list widget item
Refactors ListWidget

* Refactors ListWidget

* Adds XML for multi list widget
Refactors MultiListWidget

* Refactors LabelWidget

yanokwa committed:
Add project specific secrets and instructions for building releases (#1170)

grzesiek2010 committed:
Wrong date pickers shown for no-calendar appearances in Android 7.0 (#1171)

shobhitagarwal1612 committed:
Improves spinner widget (#1165)

* Adds spinner layout xml
Merges style.xml and styles.xml into style.xml

* uses the accent color to show selected item
removes the "remove response" from dropdown list when no item is selected

* Removes unused resources

* Adds style to the spinner
Removes tint color from "Remove Response" in drop down list

shobhitagarwal1612 committed:
Adds options menu in AdminPreferences for devices not having hardware menu button (#1158)

* Adds options menu in AdminPreferences for devices not having hardware menu button

* Modifies the AdminPreferencesActivityTest

shobhitagarwal1612 committed:
Removes sort list and filter list options from device menu button (#1175)

grzesiek2010 committed:
Code improvements

shobhitagarwal1612 committed:
Improves url widget (#1174)

* Improves url widget
Opens url in custom chrome tab by default

grzesiek2010 committed:
Force landscape orientation for draw widget (#1177)

grzesiek2010 committed:
RangeDecimalWidget (#1144)

shobhitagarwal1612 committed:
Bad submission_url in forms from Aggregate fixed (#1059)

yanokwa committed:
Using released JavaRosa 2.3.0 (#1196)

shobhitagarwal1612 committed:
show/hide the save form and form goto according to saved preferences (#1185)

shobhitagarwal1612 committed:
Fixes crash on device rotation in Server Preferences (#1188)

grzesiek2010 committed:
New class names

grzesiek2010 committed:
Added new abstract class

grzesiek2010 committed:
InstanceGoogleSheetsUploader extends InstanceUploader

grzesiek2010 committed:
Add vertical appearance to range widget (#1186)

grzesiek2010 committed:
StringWidget/IntegerWidget/DecimalWidget with  option allow to use paste option (#1193)

shobhitagarwal1612 committed:
Replaces nested preference screens with preference headers

shobhitagarwal1612 committed:
Removes preference_layout

shobhitagarwal1612 committed:
Fixes show/hide bug in non admin mode

shobhitagarwal1612 committed:
Fixes toolbar in API 19 devices

shobhitagarwal1612 committed:
Adds background color to option menu in admin settings

shobhitagarwal1612 committed:
Moves access preferences fragment classes within AdminPreferencesFragment class

shobhitagarwal1612 committed:
Refactors code

shobhitagarwal1612 committed:
Removes divider line between preference items

grzesiek2010 committed:
Tell user that PBF offline tiles are not supported (#1172)

grzesiek2010 committed:
The whole text in Settings, About is grey (#1201)

VisheshVadhera committed:
Change "autosend" to "auto send". (#1183)

nap2000 committed:
Add timing audit feature #257 (#760)

shobhitagarwal1612 committed:
Adds audit preference

lognaturel committed:
Merge pull request #1199 from shobhitagarwal1612/issue-1184

Replaces nested preference screens with preference headers

lognaturel committed:
Update translations for June 2017 (#1205)

lognaturel committed:
Remove audit setting (#1203)

* Make the default value for audit pref true.

This setting suppresses the form-level setting when false. The default should be true.

* Remove the audit setting from the user interface.

This leaves dead code pending a decision regarding the setting. If the setting is removed, some code will need to be cleaned up.

grzesiek2010 committed:
Fixed bugs with menu (#1218)

grzesiek2010 committed:
Exception java.lang.RuntimeException: An error occurred while executing doInBackground() (#1209)

lognaturel committed:
Update available translations. (#1213)

grzesiek2010 committed:
Fixed IconLocation lint (#1223)

lognaturel committed:
Add All Widgets form link (#1212)

* Add All Widgets form link

It's useful to have a direct link to both for testing and to keep up to date.

* Put All Widgets form first.

grzesiek2010 committed:
Reduce redundancy in WidgetFactory (#1187)

grzesiek2010 committed:
App crashes when user goes to URL widget on Android 4.2.1, Infinix X451 (#1224)

grzesiek2010 committed:
Unrecognized token error for column headers starting with a number (#1225)

grzesiek2010 committed:
External read-only widgets should not allow paste (#1198)

grzesiek2010 committed:
Handle readonly using all widgets (#1195)

grzesiek2010 committed:
Changed Tiber.e to Tiber.i in the discardEntityBytes method (#1232)

grzesiek2010 committed:
Ignore ContentDescription (#1230)

grzesiek2010 committed:
Exception java.net.UnknownHostException: Unable to resolve host... (#1229)

* Exception java.net.UnknownHostException: Unable to resolve host...

* Exception org.opendatakit.httpclientandroidlib.conn.ConnectTimeoutException

grzesiek2010 committed:
Crash when attempting to get device details (#1228)

shobhitagarwal1612 committed:
Adds dimens.xml to reduce magic values (#1189)

grzesiek2010 committed:
SQLiteException - SQLiteConnection.nativePrepareStatement (#1236)

grzesiek2010 committed:
Fixed bad class name

nap2000 committed:
move timerLogger object into formController. Fix #1234. (#1240)

lognaturel committed:
Merge pull request #1151 from grzesiek2010/COLLECT-791_3

Configure auto-send and auto-delete at the form level

lognaturel committed:
Outline the review process (#1215)

* Outline the review process

* Format review tags.

grzesiek2010 committed:
Check for null before dereferencing exception details (#1248)

getDetails() returns null when the response isn't json.

grzesiek2010 committed:
Name disappears on rotation (#1246)

SIMRAN1 committed:
Move Google Sheets regex to methods and add tests (#1116)

grzesiek2010 committed:
Add "no-ticks" and vertical "no-ticks" appearances for RangeWidget (#1200)

lognaturel committed:
Import Outcome from InstanceUploader base class. (#1261)

heyjamesknight committed:
Draft LocationClient.

heyjamesknight committed:
Draft LocationClient.

heyjamesknight committed:
Draft LocationClient.

heyjamesknight committed:
Draft LocationClient.

heyjamesknight committed:
LocationClients finished, documented, and tested.

heyjamesknight committed:
Draft of Activity updates.

heyjamesknight committed:
Merge branch 'issue-508-locationupdate' of github.com:PembrokeStudio/collect into issue-508-locationupdate

heyjamesknight committed:
Merge branch 'master' into issue-508-locationupdate

heyjamesknight committed:
Initial tests.

Commit:
Finished GeoPointActivity tests.

Commit:
Test for GeoPointMapActivity

Commit:
Added remaining test scaffolds.

Commit:
Major tests complete.

Commit:
Pre-PR cleanup

heyjamesknight committed:
Fixed some overzealous refactoring errors.

heyjamesknight committed:
Updated Mockito & Robolectric, fixed broken tests for Mockito 2.

heyjamesknight committed:
Missed Robolectric.

heyjamesknight committed:
Fixed discrepencies with lib update branch.

lognaturel commented:
Thanks! I've only taken a quick look so I only have surface comments for now. More soon. When the accuracy goes below the [`accuracyThreshold`](http://xlsform.org/#gps-accuracy-threshold) for the GeoPoint widget with no map, the reading should be automatically collected and the dialog closed. That's the behavior in the released version. It looks like now any reading requires user intervention.

With the currently released version, inside I have an accuracy of about 22m. When I walk outside it goes down to 13 and then to 4.55 almost immediately. With this version, I walk outside and the accuracy stays at 22m for a long time. I've tried this a couple of times and gotten the same result so it would be good to get another sanity check. This is with a Moto G4.

Things we should try on real devices:
- not having Play Services installed -- GeoPoint with no map and GeoShape with Open Street Map should still work
- slower device, compare against currently-released version

heyjamesknight commented:
@lognaturel: Looking into the location return, not sure what's going on there—that portion of the code should not have changed so I must have made a mistake somewhere. On the accuracy front, I'll modify the LocationRequest object to specifically request finer accuracy when LocationClient is started with a higher Priority. 

Sorry for not catching that sooner—I work in the old dense part of the city so GPS accuracy is always poor.

For on device testing:
- Anything without Play Services should fail the areGooglePlayServicesAvailable() test in LocationClients and receive the old style client. We can force this for testing by having this method return false.
- The oldest device I have on hand is a Nexus 5—is there anyone in the community we could pull in for testing on older devices? 

I've also updated the tests that failed with CI, they should pass now.

Let me know if you find any other issues!

heyjamesknight committed:
Fixed CI failures.

heyjamesknight committed:
Fixed other failing test.

heyjamesknight committed:
Fixed Priority issue on GoogleLocationClient. Added ability to set updateInterval.

heyjamesknight committed:
Fixed Checkstyle issues.

heyjamesknight commented:
Ah actually the accuracy issue was because we were defaulting to PRIORITY_BALANCED_POWER_ACCURACY which prefers the Network provider. Switched to PRIORITY_HIGH_ACCURACY and now things are better. I've also added the ability to set an update interval on the LocationClient which sped things up considerably.

The dialog will also automatically close when the accuracy drops below 5m. That should still be happening and is being tested in GeoPointActivityTest (from line 111 to 130).

heyjamesknight commented:
We should chat later about what we want to do re: setting update intervals on the remaining activities. We can decrease the time between updates at the cost of battery life, but I'm not sure under what conditions each of the remaining Activities is used and how best to manage that tradeoff.
